From: Fernando Fernandez Mancera <ffmancera@riseup.net>
Date: Thu, 25 Aug 2022 15:18:08 +0200
Subject: bond: fix missing assignment of lp_interval_has

The variable `lp_interval` was being assigned instead of
`lp_interval_has`. The `lp_interval` bond option was not being set
correctly.

https://bugs.launchpad.net/network-manager/+bug/1987001

Fixes: e064eb9d1361 ('bond: use netlink to set bond options')
(cherry picked from commit 7d4307e8dfdd6c8d33d35147d593259fbaf0f7bf)
(cherry picked from commit 1b704e2f425ec4199386a8530b60ffbe14898025)
---
 src/core/devices/nm-device-bond.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/devices/nm-device-bond.c b/src/core/devices/nm-device-bond.c
index 41473ab..44bb316 100644
--- a/src/core/devices/nm-device-bond.c
+++ b/src/core/devices/nm-device-bond.c
@@ -461,7 +461,7 @@ _platform_lnk_bond_init_from_setting(NMSettingBond *s_bond, NMPlatformLnkBond *p
     props->peer_notif_delay_has = (props->miimon || props->arp_interval) && props->peer_notif_delay;
     props->arp_all_targets_has  = props->arp_interval && props->arp_all_targets;
     props->resend_igmp_has      = props->resend_igmp != 1;
-    props->lp_interval          = props->lp_interval != 1;
+    props->lp_interval_has      = props->lp_interval != 1;
     props->tlb_dynamic_lb_has   = NM_IN_SET(props->mode, NM_BOND_MODE_TLB, NM_BOND_MODE_ALB);
 }
 
